test/testutilsbytearray.cpp

```diff
@@ -123,6 +124,33 @@ private slots:
         QCOMPARE(Utils::ByteArray::toBase32("0000000000"), "GAYDAMBQGAYDAMBQ");
         QCOMPARE(Utils::ByteArray::toBase32("1"), "GE======");
     }
+
+    void testUnquote() const
+    {
+        const auto test = []<typename T>()
+        {
+            QCOMPARE(Utils::ByteArray::unquote<T>({}), {});
+            QCOMPARE(Utils::ByteArray::unquote<T>("abc"), "abc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"abc\""), "abc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"a b c\""), "a b c");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"abc"), "\"abc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("abc\""), "abc\"");
+            QCOMPARE(Utils::ByteArray::unquote<T>(" \"abc\" "), " \"abc\" ");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"a\"bc\""), "a\"bc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("'abc'", "'"), "abc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("'abc'", "\"'"), "abc");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"'abc'\"", "\"'"), "'abc'");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"'abc'\"", "'\""), "'abc'");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"'abc'\"", "'"), "\"'abc'\"");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"abc'", "'"), "\"abc'");
+            QCOMPARE(Utils::ByteArray::unquote<T>("'abc\"", "'"), "'abc\"");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\"\""), "");
+            QCOMPARE(Utils::ByteArray::unquote<T>("\""), "\"");
+        };
+
+        test.template operator()<QByteArray>();
+        test.template operator()<QByteArrayView>();
```

Comment:
🙃

Reply:
That's not my idea. https://github.com/qbittorrent/qBittorrent/pull/23083#discussion_r2281007905